libraries/AP_Common/AP_FWVersionDefine.h

```diff
@@ -21,12 +21,27 @@
 #endif
 
 #include <AP_Common/AP_FWVersion.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
 
 const AP_FWVersion AP_FWVersion::fwver{
+    // Version header struct
+    .header = 0x61706677766572fb, // First 7 MSBs: "apfwver", LSB is the checksum of the previous string: 0xfb
+    .header_version = static_cast<uint16_t>(0x0100), // Major and minor version
+    .pointer_size = static_cast<uint8_t>(sizeof(void*)),
+    .reserved = 0,
+    .vehicle_type = static_cast<uint8_t>(APM_BUILD_DIRECTORY),
```

Comment:
how is APM_BUILD_DIRECTORY a uint8_t?

Reply:
The name is indeed weird, but the variable contain numeric values.
Code: https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Vehicle/AP_Vehicle_Type.h#L17-L40
Relevant comment: https://github.com/ArduPilot/ardupilot/pull/15415#discussion_r494294892

Reply:
A little bit more information; waf sets `APM_BUILD_DIRECTORY` to be (e.g.) `APM_BUILD_Rover` (the latter being a define (in the top of `AP_Vehicle_Type.h`):

```
def get_legacy_defines(sketch_name):
    return [
        'APM_BUILD_DIRECTORY=APM_BUILD_' + sketch_name,
        'SKETCH="' + sketch_name + '"',
        'SKETCHNAME="' + sketch_name + '"',
    ]
```

We really ought to rename this - remove `DIRECTORY` and tidy away `APM` while we're at it.  Perhaps `AP_BUILD_TYPE`?  That's a later thing, anyway.